car.py

```diff
@@ -1,129 +1,50 @@
-        self.elec = electronics.Electronics()
-        self.mot = motor.Motor()
-        self.prevTime = startingTime
-        self.deltaTime = 0
+    def __init__(self, track: Track, time: int) -> None:
+        max_capacity = 5400 # Wh
+        self.lon, self.lat = track.coords
+        self.temperature = 25 #update
+        self.time = time
         
-    def drive(self, speed, time, slope): #parameters required for each tick
-        self.deltaTime = time - self.prevTime
-        self.prevTime = time
-        #current provided by array
-        self.weather_conditions.pull_weather_data(time)
-        arr = aaron_array.Array(self.weather_conditions, 35)
-        # ArrayVoltage = self.voltageOfaSingeCell*242
-        #currentFromArray = arr.get_power()/ArrayVoltage
-        array_power = arr.get_power() #THIS IS IN POWER
-
-        #current drawn from electronics
-        # electronics_power = self.elec.run() #THIS IS IN CURRENT
-        electronics_power = 76.18 #power draw from constantly used systems
-    
-        #current drawn by motor
-        self.mot.updateParameters(speed, slope, self.deltaTime)
-        motor_current = self.mot.currentMotor() #CURRENT
-        motor_power = motor_current*96 #?!?!?!?! what is multiplier of 96????
-        if motor_power < 0:
-            motor_power = 0
-        max_speed = self.mot.dynamics.max_velocity(abs(slope))
-
-        #total_current send to battery (- is discharge, + is charge)
-        total_power = (array_power - (motor_power + electronics_power))*self.deltaTime/3600 #watt hours
+        self.battery = battery.Battery(max_capacity)
+        self.array = solar_array.Array(self, self.temperature)
+        self.motor = motor.Motor()
 
-        print(f'{total_power}')
-
-        #calculate capacity loss and SOC of battery pack
-        # total_pack_capacity = 5400 #parameter needs to be verified
-        # batteryPack = battery_pack.BatteryPack(self.capacity/432,total_power)
-        # batteryPack.updateModuleInternalResistance()
-        # self.capacity += batteryPack.LostCapacity(time) #is batteryCharge the same thing as capactiy?
-        # battery_charge = (self.capacity/total_pack_capacity)*100
+    def drive(self, speed, slope, time):
+        print("Car is driving")
+        deltaTime = time - self.time
         
-        self.capacity += total_power
-
-        battery = self.capacity/self.totalCapacity*100
-        if battery < 0:
-            battery = 0
-        elif battery > 100:
-            battery = 100
-
-        print('Power draw from motor: {0:.2f} W'.format(motor_power))
-        print('Power provided from array: {0:.2f} W'.format(array_power))
-        print('Power draw from electronics {0:.2f} W'.format(electronics_power))
-        print('Battery is at {0:.2f}%'.format(battery))
-        print('Battery capacity is {0:.2f}% W', format(self.capacity))
+        arrayPower = self.array.get_power()
+        electronicsPower = 77 #power draw from constantly used systems
```

Comment:
again make this a defined constant. in general, avoid "magic numbers"

Reply:
also if this isn't actually going to change during runtime, you don't need to make it a variable at all